tests/parsers/test.json.js

```diff
@@ -0,0 +1,257 @@
+  });
+
+  // Chrome will accept multiline /* */ comments, but Firefox will not and
+  // the Web Extension spec does not allow them. So we will error on them.
+  it('does not parse JSON with a multiline comment', () => {
+    var addonLinter = new Linter({_: ['bar']});
+    var json = `/* I am a JSON comment, sigh*/\n${validManifestJSON()}`;
+    var jsonParser = new JSONParser(json, addonLinter.collector);
+    jsonParser.parse();
+
+    assert.equal(jsonParser.isValid, false);
+    var errors = addonLinter.collector.errors;
+    // There should not be another error; a file with block-level comments
+    // will throw that specific error and not a parse error.
+    assert.lengthOf(errors, 1);
+    assert.equal(errors[0].code, messages.JSON_BLOCK_COMMENTS.code);
+    assert.equal(errors[0].message, messages.JSON_BLOCK_COMMENTS.message);
+  });
+
+  it('parses the example from Chrome developer docs', () => {
+    var addonLinter = new Linter({_: ['bar']});
+    // Example from https://developer.chrome.com/extensions/manifest
+    var json = [
+      '{',
+      '// Required',
+      '"manifest_version": 2,',
+      '"name": "My Extension",',
+      '// Make the hell sure to use semvar.org if increasing this',
+      '"version": "0.0.1"',
+      '}',
+    ].join('\n');
+    var jsonParser = new JSONParser(json, addonLinter.collector);
+    jsonParser.parse();
+
+    assert.equal(jsonParser.isValid, true);
+    assert.notInclude(jsonParser._jsonString, 'semvar.org');
+  });
+
+  it('returns the correct error for malformed JSON', () => {
+    var addonLinter = new Linter({_: ['bar']});
+    var json = `{"something": true,\n// I am a JSON comment, sigh\nblah}`;
+    var jsonParser = new JSONParser(json, addonLinter.collector);
+    jsonParser.parse();
+
+    assert.equal(jsonParser.isValid, false);
+    var errors = addonLinter.collector.errors;
+    assert.equal(errors[0].code, messages.JSON_INVALID.code);
+    assert.include(errors[0].message, 'Your JSON is not valid.');
+  });
+
+  it("doesn't evaluate JS code in comments", () => {
+    var addonLinter = new Linter({_: ['bar']});
+    var json = '// eval("");\n{"something": true}\nvar bla = "foo";';
+    var jsonParser = new JSONParser(json, addonLinter.collector);
+    jsonParser.parse();
+
+    assert.equal(jsonParser.isValid, false);
+    var errors = addonLinter.collector.errors;
+    assert.equal(errors[0].code, messages.JSON_INVALID.code);
+    assert.include(errors[0].message, 'Your JSON is not valid.');
+    assert.notInclude(jsonParser._jsonString, 'var bla');
+    assert.notInclude(jsonParser._jsonString, 'eval');
+  });
+
+  it("doesn't evaluate JS code even though esprima is used", () => {
+    var addonLinter = new Linter({_: ['bar']});
+    var json = [
+      '{',
+      '// Required',
+      '"manifest_version": 2,',
+      '"name": "My Extension",',
+      '// Make the hell sure to use semvar.org if increasing this',
+      '"version": eval("alert(\'uh-oh\')")',
```

Comment:
I see this is a moved test and not new, but if the alert was fired, would it fail the test?